centreon/src/Core/Service/Infrastructure/Repository/DbReadServiceRepository.php

```diff
@@ -0,0 +1,145 @@
+use Utility\SqlConcatenator;
+
+class DbReadServiceRepository extends AbstractRepositoryRDB implements ReadServiceRepositoryInterface
+{
+    use LoggerTrait;
+
+    /**
+     * @param DatabaseConnection $db
+     */
+    public function __construct(DatabaseConnection $db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function exists(int $serviceId): bool
+    {
+        $concatenator = $this->getServiceRequestConcatenator();
+
+        return $this->existsService($concatenator, $serviceId);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function existsByAccessGroups(int $serviceId, array $accessGroups): bool
+    {
+        $accessGroupIds = array_map(
+            static fn(AccessGroup $accessGroup) => $accessGroup->getId(),
+            $accessGroups
+        );
+
+        $concatenator = $this->getServiceRequestConcatenator($accessGroupIds);
+
+        return $this->existsService($concatenator, $serviceId);
+    }
+
+    public function findMonitoringServerId(int $serviceId): int
+    {
+        $request = $this->translateDbName(<<<'SQL'
+            SELECT id
+            FROM `:db`.`nagios_server` ns
+            INNER JOIN `:db`.`ns_host_relation` nshr
+                ON nshr.nagios_server_id = ns.id
+            INNER JOIN `:db`.`host_service_relation` hsr
+                ON hsr.host_host_id = nshr.host_host_id
+            WHERE hsr.service_service_id = :service_id
+            SQL
+        );
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':service_id', $serviceId, \PDO::PARAM_INT);
+        $statement->execute();
+
+        return (int) $statement->fetchColumn();
+    }
+
+    /**
+     * @param int[] $accessGroupIds
+     *
+     * @return SqlConcatenator
+     */
+    private function getServiceRequestConcatenator(array $accessGroupIds = []): SqlConcatenator
+    {
+        $concatenator = (new SqlConcatenator())
+            ->defineFrom(
+                <<<'SQL'
+                    FROM
+                        `:db`.`service` s
+                    SQL
+            );
+
+        if ([] !== $accessGroupIds) {
+            // TODO: will be handled later
+        }
+
+        return $concatenator;
+    }
+
+    /**
+     * @param SqlConcatenator $concatenator
+     * @param int $serviceId
+     *
+     * @throws \PDOException
+     *
+     * @return bool
+     */
+    private function existsService(SqlConcatenator $concatenator, int $serviceId): bool
+    {
+        $concatenator
+            ->defineSelect(
+                <<<'SQL'
+                    SELECT 1
+                    SQL
+            )
+            ->appendWhere(
+                <<<'SQL'
+                    WHERE s.service_id = :service_id
+                    SQL
+            )
+            ->appendWhere(
+                <<<'SQL'
+                    WHERE s.service_register = '1'
+                    SQL
+            )
```

Comment:
More understandable than two concatenated WHERE conditions, don't you think?
```suggestion
            ->appendWhere(
                <<<'SQL'
                    WHERE s.service_id = :service_id AND s.service_register = '1'
                    SQL
            )
```